src/node_url.cc

```diff
@@ -1545,44 +1546,61 @@ void URL::Parse(const char* input,
 }  // NOLINT(readability/fn_size)
 
 // https://url.spec.whatwg.org/#url-serializing
-std::string URL::SerializeURL(const struct url_data* url,
+std::string URL::SerializeURL(const url_data& url,
                               bool exclude = false) {
-  std::string output = url->scheme;
-  if (url->flags & URL_FLAGS_HAS_HOST) {
+  std::string output;
+  output.reserve(
+    10 + // We generally insert < 10 separator characters between URL parts
```

Comment:
To address the linter error

```suggestion
    10 +  // We generally insert < 10 separator characters between URL parts
```